test/parallel/test-whatwg-url-override-hostname.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const { URL } = require('url');
+
+{
+  const url = new (class extends URL { get hostname() { return 'bar.com'; } })('http://foo.com/');
+  assert.strictEqual(url.href, 'http://foo.com/');
+  assert.strictEqual(url.toString(), 'http://foo.com/');
+  assert.strictEqual(url.toJSON(), 'http://foo.com/');
+  assert.strictEqual(url.hostname, 'bar.com');
```

Comment:
It would likely be good to expand the test to cover the other property accessors to make sure those continue to match browser behavior also.

Reply:
Nice, the code has been updated